gen.go

```diff
@@ -0,0 +1,215 @@
+package avro
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"io"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+type GenConf struct {
+	PackageName string
+	OutFileName string
+}
+
+var primitiveMappings = map[Type]string{
+	"string":  "string",
+	"bytes":   "[]byte",
+	"int":     "int",
+	"long":    "int64",
+	"float":   "float32",
+	"double":  "float64",
+	"boolean": "bool",
+}
+
+func GenerateFrom(gc GenConf, s string) (io.Reader, error) {
+	schema, err := Parse(s)
+	if err != nil {
+		return nil, err
+	}
+
+	rSchema, ok := schema.(*RecordSchema)
+	if !ok {
+		return nil, errors.New("can only generate Go code from Record Schemas")
+	}
+	file := ast.File{
+		Name: &ast.Ident{Name: strcase.ToSnake(gc.PackageName)},
+	}
+	generateFrom(rSchema, &file)
+
+	buf := &bytes.Buffer{}
+	if err = writeFile(buf, &file); err != nil {
+		return nil, err
+	}
+	return buf, nil
+}
+
+func generateFrom(schema Schema, acc *ast.File) string {
+	switch t := schema.(type) {
+	case *RecordSchema:
+		typeName := strcase.ToCamel(t.Name())
```

Comment:
The name could be fully namespaced, and it may not be desired to include the namespace. Perhaps we should have an option to remove it if it exists. Not sure what a good default would be.

Reply:
My use cases over the last year with avro + Go have always been to use just the name of the record as the struct. Mostly the namespaces I deal with are of the form `org.name.domain.Entity`, so the only thing I would care about here is generating a struct with the name of the record and ignore the namespace `org.name.domain.Entity`.

Still, my use case isn't the only one out there. I could add an option `PrefixStructsWithNamespace`? And I'm assuming we would remove the `.` in the namespace.

Could be deffered to when the need arises tho. Up to you.

Reply:
Happy to defer until later. Was just off the top of my head.